### PR TITLE
Adding automated localhost benchmark for OSX

### PR DESCRIPTION
**Pull request by jabrena**
Hi @spencergibb,

Using your notes from `README`, I have created a Script to test in `OSX` the whole process.
I can confirm your notes in a relative way.

The new `Spring Gateway` has improved the performance but it is true that `Linkerd` has a very good performance too but in all tests, the new gateway has the best results.

It could be interesting to deploy in `CF` the different ways to have a clean benchmark but in local, the results could orient the community.

It could be interesting to compare features between the new `Spring Gateway` and `Linkerd` in an Article.

Juan Antonio

### CODE REVIEW

File: benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,212 @@
+#!/bin/bash
+# A Bash script to execute a Benchmark about implementation of Gateway pattern for Spring Cloud
+
+echo "Gateway Benchmark Script"
+
+OSX="OSX"
+WIN="WIN"
+LINUX="WIN"
```
Comment (Fydon) [inline benchmark.sh]: Should this be `LINUX="LINUX"`?

**Comment by jabrena** [comment]
Hi @Fydon,

You have reason.
I will upgrade the Script today.

**Comment by spencergibb** [comment]
I also run linux, so I couldn't test this.

**Comment by jabrena** [comment]
Fixed minor change

**Comment by spencergibb** [comment]
@jabrena thanks for the contribution!

**Comment by jabrena** [comment]
This weekend, I will try to move to `Jmeter`.

I tried to improve the `wrk` solution but the technical support is not fast.
https://github.com/wg/wrk/issues/315

So, I will create a feature branch in my local repo to start with this process.
The idea is to reduce the benchmark execution time with a dynamic way to determinate the best `jvm warnup` for any gateway.

Later with a stable automated benchmark process, I will try to create a stage on Spring Cloud Gateway repo to automate the process to guarantee that the new `Spring Gateway` offers better performance than `Netflix Zuul`.

Juan Antonio

**Comment by spencergibb** [comment]
Sounds good

**Pull request merged**